source/math/FloatMatrix.ooc

```diff
@@ -94,6 +94,19 @@ FloatMatrix : cover {
 	}
 
 	// <summary>
+	// Calculates the trace of a square matrix.
+	// </summary>
+	// <returns>The trace of the matrix.</returns>
+	trace: func -> Float {
+		if (!this isSquare)
+			InvalidDimensionsException new() throw()
```

Comment:
It would be better to get rid of `InvalidDimensionsException` and use `raise("Invalid dimensions in FloatMatrix trace: not square!")` or something.